src/gs-discovery-feed-content-provider.c

```diff
@@ -136,6 +130,38 @@ get_app_thumbnail_cached_filename (GsApp *app)
 	return NULL;
 }
 
+static gboolean
+filter_for_discovery_feed_apps (GsApp *app, gpointer user_data)
+{
+	g_autofree gchar *app_thumbnail_filename = NULL;
+	GdkPixbuf *pixbuf;
+	g_autoptr(GVariant) icon_serialized = NULL;
+
+	if (gs_app_get_state (app) != AS_APP_STATE_AVAILABLE)
```

Comment:
Would it be possible to express this constraint when doing the search instead of at filtering time?

Reply:
Doesn't look like it - or at least, the shell search provider also needs to make such a check: https://github.com/endlessm/gnome-software/blob/master/src/gs-shell-search-provider.c#L96